pyfarm/agent/sysinfo/disks.py

```diff
@@ -0,0 +1,78 @@
+# No shebang line, this module is meant to be imported
+#
+# Copyright 2015 Ambient Entertainment GmbH & Co. KG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Disks
+-----
+
+Contains information about the local disks.
+"""
+
+try:
+    from os import statvfs
+except ImportError:  # pragma: no cover
+    statvfs = NotImplemented
+
+try:
+    from wmi import WMI
+except ImportError:  # pragma: no cover
+    WMI = NotImplemented
+
+from pyfarm.core.enums import WINDOWS, LINUX, BSD
+from pyfarm.agent.logger import getLogger
+
+logger = getLogger("agent.disks")
+
+
+class DiskLookupError(Exception):
+    pass
+
+
+class DiskInfo:
+    def __init__(self, mountpoint, free, size):
+        self.mountpoint = mountpoint
+        self.free = free
+        self.size = size
+
+
+def disks():
```

Comment:
Can we use `psutil` instead here since it's cross-platform and will be less work?  The below for example won't account for bind mounts and may also have trouble on Linux for some non-standard setups.

```python
from collections import namedtuple
DiskInfo = namedtuple("DiskInfo", ("mountpoint", "free", "size"))

for partition in psutil.disk_partitions():
    usage = psutil.disk_usage(partition.mountpoint)
    
    info = DiskInfo(
         mountpoint=usage.mountpoint,
         free=usage.free,
         size=usage.total
    )
    
```